packages/node_modules/samples/browser-meetings/app.js

```diff
@@ -205,3 +206,212 @@ function leaveMeeting(meetingId) {
+        console.log('MeetingControls#pauseRecording() :: unable to pause recording.');
+        console.error(error);
+      });
+  }
+}
+
+
+function stopRecording() {
+  const meeting = getSelectedMeeting();
+
+  if (meeting) {
+    console.log('MeetingControls#stopRecording()');
+    generalControlsRecStatus.innerText = 'Stop recording meeting...';
+    meeting.stopRecording()
+      .then(() => {
+        generalControlsRecStatus.innerText = 'Recording stopped successfully!';
+        console.log('MeetingControls#stopRecording() :: meeting recording stopped!');
+      })
+      .catch((error) => {
+        generalControlsRecStatus.innerText = 'Error! See console for details.';
+        console.log('MeetingControls#stopRecording() :: unable to stop recording!');
+        console.error(error);
+      });
+  }
+}
+
+
+function sendDtmfTones() {
+  const meeting = getSelectedMeeting();
+  const tones = generalControlsDtmfTones.value || '';
+
+  if (!tones) {
+    console.log('MeetingControls#sendDtmfTones() :: Error, empty string.');
+    generalControlsDtmfStatus.innerText = 'Please enter DTMF tones and try again.';
+
+    return;
+  }
+
+  if (meeting) {
+    console.log('MeetingControls#sendDtmfTones()');
+    meeting.sendDTMF(tones)
+      .then(() => {
+        generalControlsDtmfStatus.innerText = 'DTMF tones sent successfully!';
+        console.log('MeetingControls#sendDtmfTones() :: DTMF tones sent!');
+      })
+      .catch((error) => {
+        generalControlsDtmfStatus.innerText = 'Error! See console for details.';
+        console.log('MeetingControls#sendDtmfTones() :: unable to send DTMF tones!');
+        console.error(error);
+      });
+  }
+}
+
+/*
+ * Utilizing a custom event listener as a way to
+ * separate code and avoid having to bloat initWebex()
+ * function with code that depends on webex being
+ * defined, initialized and ready.
+ *
+ */
+document.addEventListener('webex-ready', () => {
+  /*
+   * Using webex meeting events "meeting:added" and "meeting:removed"
+   * to update list of available meetings a user can select.
+   */
+  webex.meetings.on('meeting:added', (meeting) => {
+    updateMeetingDropdown();
+
+    // Enable the General Controls form if a meeting is available.
+    generalControlsForm.querySelectorAll('fieldset')
+      .forEach((fieldset) => { fieldset.removeAttribute('disabled'); });
+  });
+
+  webex.meetings.on('meeting:removed', (meeting) => {
+    const noMeetingsAvailable = !Object.keys(webex.meetings.getAllMeetings()).length;
+
+    updateMeetingDropdown();
+    // Disable form if no meetings are available.
+    console.log('fieldsets', generalControlsForm.querySelectorAll('fieldset'));
+    if (noMeetingsAvailable) {
+      generalControlsForm.querySelectorAll('fieldset')
+        .forEach((fieldset) => { fieldset.setAttribute('disabled', true); });
+    }
+  });
+  // {once: true} ensures the listener fires only once
+}, {once: true});
```

Comment:
For populating the `<select />` element, we should utilize `webex.meetings.getAllMeetings()` for an onclick method, as seen in an earlier section of this app.js file: `collectMeetings()`.

Additionally, we should avoid adding the logic for disabling/enabling these form elements based on the authentication state. This allows users to try the buttons in the case that no meeting was selected/loaded/available, giving them visibility of the error given in that use case.

Removing these listeners will improve the coupling [decoupling] of this sample, improving consumability of code snippets located within this file.

Reply:
I completely understand and see your point. Although, what you are describing is currently in the current meeting sample and I did not find it advantageous or helpful. 

I grew frustrated initially because typically with UX, if a button is presented it will usually perform the action labeled. It wasn't obvious in the UI that only certain buttons performed the actions given, while other buttons performed the action only if you follow the steps in a specific order. 

I was hoping to alleviate some that by disabling the button until it can perform the stated action.

Reply:
@abailey-wwt the scope of this ticket is to just rearrange the developer playground (my new name for this definitely not kitchen sink app). We have a new full on kitchen sink that we are developing that this would be awesome for. 

That being said, I hate to remove nice working code, so since @InteractiveTimmy is the architect of this, I'll let him decide on it.